cypress/support/commands.ts

```diff
@@ -23,3 +33,20 @@ Cypress.Commands.add('useSubscriptionStatus', (condition) => {
         cy.reload()
     }
 })
+
+Cypress.Commands.add('pollUntilPresent', (eventName: string, timeout = 5000): Cypress.Chainable<CaptureResult[]> => {
+    return cy.window().then({ timeout }, (win: WindowWithCypressCaptures) => {
+        return new Cypress.Promise<CaptureResult[]>((resolve, reject) => {
+            const checkEvents = (): void => {
+                const events = win._cypress_posthog_captures || []
+                if (events.some((event) => event.event === eventName)) {
+                    resolve(events) // Return the captures array
+                } else {
+                    setTimeout(checkEvents, 100) // Retry in 100ms
+                }
+            }
+            checkEvents()
+            setTimeout(() => reject(new Error(`Timed out waiting for event: ${eventName}`)), timeout)
```

Comment:
Maybe Cypress will help us here, but I see some `timeout`s leaking here. We need some `clearTimeout` hygiene here.

If we resolve inside `checkEvents` we'll still trigger the `setTimeout` in line 49, which will then be rejected eventually. Does Cypress handle this well?

On the other hand, if we reject it, we won't have cleaned up the `setTimeout` inside `checkEvents` and it'll actually keep running forever - until the Node process is done.

Reply:
fair... I think Cypress is already so heavy that we wouldn't notice 🙈 

will bear this in mind as we migrate to playright though!

Reply:
If tests pass, it does it's job 😅